File: Java/383_RansomNote.java
// 383 - Ransom Note

public boolean canConstruct(String ransomNote, String magazine) {
    for(int i = 0; i < ransomNote.length(); i++) {
        int idx = magazine.indexOf(ransomNote.charAt(i));
        if(idx == -1) { return false; }
        magazine = magazine.substring(0, idx) + magazine.substring(idx + 1);
    }
    return true;
}

File: CPP/383_RansomNote.cpp
// 383 - Ransom Note

bool canConstruct(string ransomNote, string magazine) {
    for(int i = 0; i < ransomNote.size(); i++) {
        int idx = magazine.find(ransomNote[i]);
        if(idx == -1) { return false; }
        magazine = magazine.substr(0, idx) + magazine.substr(idx + 1);
    }
    return true;
}